.github/workflows/vercel-production.yml

```diff
@@ -0,0 +1,43 @@
+name: Build and Deploy to Vercel (Production)
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build-and-deploy:
+    name: Build and Deploy to Vercel (Production)
+    runs-on: ubuntu-latest
+
+    env:
+      VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+      VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+
+    environment:
+      name: vercel-production
+      url: ${{ steps.deploy.outputs.url }}
+
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v3
```

Comment:
IIRC all GitHub actions should be pinned to specific commits for security purposes. At least in the main repo, I don't know about the website. But since this action appears to have access to secret tokens, it might make sense.